### PR TITLE
Update URL tests to reject hostnames that end in numbers but are not IPv4 IPs

### PR DESCRIPTION
**Pull request by MattMenke2**
This pull request corresponds to the proposed URL spec changes in https://github.com/whatwg/url/pull/619

It also adds a pair of tests for cases that don't seem to be covered by existing tests (A normal IPv4 IP, and a normal IPv4 IP ending in an extra dot)

I moved some pre-existing tests around that were affected by this change, but am also happy to just leave them where they are.

### CODE REVIEW

File: url/resources/urltestdata.json [hunk 1]
```diff
@@ -5161,6 +5146,36 @@
...
+    "port": "",
+    "pathname": "/",
+    "search": "",
+    "hash": ""
+  },
+  {
+    "input": "http://1.2.3.4./",
+    "base": "http://other.com/",
+    "href": "http://1.2.3.4/",
+    "origin": "http://1.2.3.4/",
```
Comment (domenic) [inline url/resources/urltestdata.json:5168]:
```suggestion
    "origin": "http://1.2.3.4",
```

File: url/resources/urltestdata.json [hunk 2]
```diff
@@ -5161,6 +5146,36 @@
     "hash": "#foo%60bar"
   },
   "# IPv4 parsing (via https://github.com/nodejs/node/pull/10317)",
+  {
+    "input": "http://1.2.3.4/",
+    "base": "http://other.com/",
+    "href": "http://1.2.3.4/",
+    "origin": "http://1.2.3.4/",
```
Comment (domenic) [inline url/resources/urltestdata.json:5153]:
```suggestion
    "origin": "http://1.2.3.4",
```

**Review by domenic — CHANGES_REQUESTED** [review]
After implementing the proposed spec update in jsdom/whatwg-url, things mostly pass but there's a couple extra slashes.

**Review by annevk — APPROVED** [review]
This looks good to me (though @domenic probably wants to run them through whatwg-url again), but having read the change proposal I was thinking we might want to cover these scenarios as well:

* Various domains ending in ".09" and ".09." (always failure as far as I can tell)
* Domains such as "test.0." and generally more dot-at-the-end variants

**Review by domenic — APPROVED** [review]
Passes in modified whatwg-url so LGTM. I'll be happy to re-run the tests if you add more cases as well.

**Comment by MattMenke2** [comment]
> This looks good to me (though @domenic probably wants to run them through whatwg-url again), but having read the change proposal I was thinking we might want to cover these scenarios as well:
> 
> * Various domains ending in ".09" and ".09." (always failure as far as I can tell)
> * Domains such as "test.0." and generally more dot-at-the-end variants

I've beefed up tests a bit.  Let me know what you think.

I've run url-constructor.any.html against Chromium hacked up to (hopefully) obey the new rules, and all the new/modified cases pass - I did that before, too, just didn't run the origin tests.  This time I've run the origin tests, too, and they also look fine.

**Review by annevk — APPROVED** [review]
Thanks! @domenic will you confirm and merge?

**Review by domenic — APPROVED** [review]

**Pull request merged**